cosmic_ray/util.py

```diff
@@ -64,5 +64,7 @@ def build_mutations(ops, to_ops):
     return [
         (idx, to_op)
         for idx, from_op in enumerate(ops)
-        for to_op in to_ops(from_op)
+        # note: mutations to self are excluded.
+        # None is a special mutation meaning to delete the operator!
+        for to_op in to_ops(from_op) if to_op is None or not isinstance(from_op, to_op)
```

Comment:
What is the `top_op is None` clause for? Is there some case where the to- and from-ops are both None?